numpy/core/src/multiarray/descriptor.c

```diff
@@ -606,40 +606,33 @@ _convert_from_array_descr(PyObject *obj, int align)
 static PyArray_Descr *
 _convert_from_list(PyObject *obj, int align)
 {
-    int n, i;
-    int totalsize;
-    PyObject *fields;
-    PyArray_Descr *conv = NULL;
-    PyArray_Descr *new;
-    PyObject *key, *tup;
-    PyObject *nameslist = NULL;
```

Comment:
Same here, `namelist` and `conv` are part of the `fail` label, and must be specifically initialized to `NULL`.

Reply:
As long as `goto fail` appears after the definitions (which it does), then there's no issue.

I think you meant `fields` not `conv`: https://github.com/numpy/numpy/pull/15288/files#diff-475dc2a28ab28625b0f02d7f006683abR669-R671